### PR TITLE
Use specs actor's MinerNominalPowerMeetsConsensusMinimum

### LINKED ISSUE #2388: Account for power in small networks correctly
There's a minimum miner size (in power) for miner's to be eligible to win blocks. When no or few miners meet the size requirement, power accounting must ignore it else we might end up with zero or one eligible miners (despite many having at least some proven storage).

In https://github.com/filecoin-project/specs-actors/pull/615 the power actor accounts for power correctly both with and without miners meeting the minimum, providing both `TotalQABytesCommitted` and `TotalQualityAdjPower` (which only counts miners above the threshold). When enforcing the minimum miner size, a node needs to use the right denominator else the sum of the power of miners might not match the denominator used.

The actors provides the `power.State.MinerNominalPowerMeetsConsensusMinimum` function which encapsulates the decision logic of whether a miner is eligible at all (https://github.com/filecoin-project/specs-actors/pull/634). 

- check eligibility with MinerNominalPowerMeetsConsensusMinimum
- divide miner QA power by the right denominator, depending on whether `st.MinerAboveMinPowerCount > ConsensusMinerMinMiners`

Note: this code doesn't necessarily live in the actors project, so please let us know if you prefer to copy it out for rework so we can drop this copy.

FYI @ZenGround0

### PR DESCRIPTION
**Pull request by arajasek**
Fixes #2388

### CODE REVIEW

**Comment by arajasek** [comment]
> Could I have this check also in the block validator please? (We are already querying the miner state to check if it exists so this shouldn't be that much expensive I suppose.)

Um, what's the block validator?

**Comment by schomatis** [comment]
> Um, what's the block validator?

https://github.com/filecoin-project/lotus/blob/c0f0e2ba45cfd36614ba598641b2cf2ea16fbfac/chain/sub/incoming.go#L142

https://github.com/filecoin-project/lotus/blob/c0f0e2ba45cfd36614ba598641b2cf2ea16fbfac/chain/sub/incoming.go#L187-L188

Inside we query its state to get the worker key and check if it matches the block, we can also query if it has enough power there:

https://github.com/filecoin-project/lotus/blob/c0f0e2ba45cfd36614ba598641b2cf2ea16fbfac/chain/sub/incoming.go#L319-L334

File: chain/sub/incoming.go [hunk 1]
```diff
@@ -203,6 +203,31 @@ func (bv *BlockValidator) Validate(ctx context.Context, pid peer.ID, msg *pubsub
 		}
 	}
 
+	// we check that the miner met the minimum power at the lookback tipset
+
+	baseTs, err := bv.stmgr.ChainStore().LoadTipSet(types.NewTipSetKey(blk.Header.Parents...))
```
Comment (magik6k) [inline chain/sub/incoming.go:208]: What if we don't have parents synced (e.g. got a small fork)?

Comment (schomatis) [reply chain/sub/incoming.go:208]: We derive this information from our heaviest tipset,

https://github.com/filecoin-project/lotus/blob/c0f0e2ba45cfd36614ba598641b2cf2ea16fbfac/chain/sub/incoming.go#L297-L298

We probably should include this validation as part of `getMinerWorkerKey` and do it all there (and if it fails we can check if `isChainNearSynced` before rejecting).

File: chain/sub/incoming.go [hunk 2]
```diff
@@ -283,59 +283,82 @@ func (bv *BlockValidator) validateMsgMeta(ctx context.Context, msg *types.BlockM
...
+		bv.keycache[addr.String()] = key
+		bv.mx.Unlock()
 	}
-	aso := blk.RawData()
 
-	var mst miner.State
-	err = mst.UnmarshalCBOR(bytes.NewReader(aso))
+	// we check that the miner met the minimum power at the lookback tipset
+
+	baseTs, err := bv.stmgr.ChainStore().LoadTipSet(types.NewTipSetKey(bh.Parents...))
```
Comment (schomatis) [inline chain/sub/incoming.go:340]: The same issue raised by @magik6k applies here the same, we should be picking the heaviest tipset not the parent of whatever we received. Anyway, this was my fault for derailing the original intent of this PR, will submit something later this week.

**Pull request merged**